lib/manager/pip_setup/extract.ts

```diff
@@ -1,139 +1,79 @@
-    const stderr = res.stderr
-      .replace(/.*\n\s*import imp/, '')
-      .trim()
-      .replace('fatal: No names found, cannot describe anything.', '');
-    if (stderr.length) {
-      logger.warn({ stdout: res.stdout, stderr }, 'Error in read setup file');
-    }
-  }
-  return parseReport(packageFile);
+    datasource: PypiDatasource.id,
+  };
+
+  return { ...ctx, deps: [...ctx.deps, dep] };
+}
+
+function depSkipHandler(ctx: Context): Context {
+  const dep = ctx.deps[ctx.deps.length - 1];
+  const deps = ctx.deps.slice(0, -1);
+  deps.push({ ...dep, skipReason: SkipReason.Ignored });
+  return { ...ctx, deps };
 }
 
-export async function extractPackageFile(
+const incompleteDepString = q
+  .str<Context>(new RegExp(cleanupNamedGroups(depPattern)))
+  .op(/^\+|\*$/);
+
+const depString = q
+  .str<Context>(new RegExp(cleanupNamedGroups(depPattern)), depStringHandler)
+  .opt(
+    q
+      .opt(q.op<Context>(','))
+      .comment(/^#\s*renovate\s*:\s*ignore\s*$/, depSkipHandler)
+  );
+
+const query = q.alt(incompleteDepString, depString);
+
+export function extractPackageFile(
   content: string,
-  packageFile: string,
-  config: ExtractConfig
-): Promise<PackageFile | null> {
-  logger.debug('pip_setup.extractPackageFile()');
-  let setup: PythonSetup;
-  try {
-    setup = await extractSetupFile(content, packageFile, config);
-  } catch (err) {
-    logger.debug({ err, content, packageFile }, 'Failed to read setup.py file');
-  }
-  if (!setup) {
-    return null;
-  }
-  const requires: string[] = [];
-  if (setup.install_requires) {
-    requires.push(...setup.install_requires);
-  }
-  if (setup.extras_require) {
-    for (const req of Object.values(setup.extras_require)) {
-      requires.push(...req);
-    }
-  }
-  const regex = new RegExp(`^${dependencyPattern}`);
-  const lines = content.split('\n');
-  const deps = requires
-    .map((req) => {
-      const lineNumber = lines.findIndex((l) => l.includes(req));
-      if (lineNumber === -1) {
-        return null;
-      }
-      const rawline = lines[lineNumber];
-      let dep: PackageDependency = {};
-      const [, comment] = rawline.split('#').map((part) => part.trim());
-      if (isSkipComment(comment)) {
-        dep.skipReason = SkipReason.Ignored;
-      }
-      regex.lastIndex = 0;
-      const matches = regex.exec(req);
-      if (!matches) {
-        return null;
-      }
-      const [, depName, , currentValue] = matches;
-      dep = {
-        ...dep,
-        depName,
-        currentValue,
-        managerData: { lineNumber },
-        datasource: PypiDatasource.id,
-      };
-      return dep;
-    })
-    .filter(Boolean)
-    .sort((a, b) =>
-      a.managerData.lineNumber === b.managerData.lineNumber
-        ? a.depName.localeCompare(b.depName)
-        : a.managerData.lineNumber - b.managerData.lineNumber
-    );
-  if (!deps.length) {
-    return null;
-  }
-  return { deps };
+  _packageFile: string,
+  _config: ExtractConfig
```

Comment:
Remove them if unused? I'm surprised you don't need `packageFile` though - is there no need to look in files other than `setup.py`?

Reply:
Leads to CI error, so leaving it with underscored names